Switch to Visual Studio 2015

chfast opened the pull request.

debris commented:
I can't run it (do not have windows vm), but everything looksgood 👍

bobsummerwill commented on build_cryptopp.bat:
```diff
@@ -1,11 +1,10 @@
 REM clone cryptopp
-if not exist build\cryptopp git clone -q https://github.com/debris/cryptopp build\cryptopp
+if not exist build\cryptopp git clone -q https://github.com/chfast/cryptopp build\cryptopp
```
Please could you pull that cryptopp repo into https://github.com/ethereum organization?   Thanks!
I started looking at that myself in doublethinkco, but we should just pull it into ethereum org.

bobsummerwill commented:
Looking good, though!   I assume you'll squash when it's all working?

Note - the Windows build machine within our automation is running Windows 8.1 Home edition, and that doesn't support RDC, so only people in the office would be able to do a VS2015 install on it.

@chriseth, @yann300, @LianaHus Please can one of you start installing VS 2015 Community Edition Update 2 Desktop on that machine?   Thanks!

bobsummerwill commented:
So yeah - needs squashing, and the CryptoPP should be moved into ethereum organization.

And we'll need to coordinate with installing VS2015 on the build machine.

Other than that, looks good to me.

chfast commented:
Why squashing?

On Fri, May 13, 2016, 23:05 Bob Summerwill notifications@github.com wrote:

> So yeah - needs squashing, and the CryptoPP should be moved into ethereum
> organization.
> 
> And we'll need to coordinate with installing VS2015 on the build machine.
> 
> Other than that, looks good to me.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/ethereum/cpp-dependencies/pull/4#issuecomment-219159003

bobsummerwill commented:
"Why squashing".   That's what we do as a matter of course with our PRs.    Because the interim steps aren't really important, and just make the history noisy.

I don't personally care, but @chriseth does.

chfast commented:
Can you review that again?

I can squash it on merge if you want (GitHub feature).

bobsummerwill commented:
All looks great to me!

I am personally fine with squash-on-merge, but @chriseth doesn't like it for a subtle reason which escapes me right now.   If you go digging on https://gitter.im/ethereum/GPP you might find the thread about that.    Or just ask him.

chfast committed:
Update build scripts to use Visual Studio 2015

chfast committed:
Add CMake scripts for building dependencies

Add CMake scripts and helper functions to build this dependencies:
- boost (ported from .bat script, version bump to 1.61),
- LLVM,
- OpenCL ICD Loader.

Also added install_deps.cmake scripts that downloads all dependencies (including Qt) from GitHub release page.

chfast commented:
Squashed.
